fix: FileWidget customRequest support

davecoates opened the pull request:
- Properly support customRequest along with progress updates
- Fix bug when multiple={true} and limit={1} (limit wasn't enforced)
- Fix issue where preview sometimes was not generated when it should have been

davecoates committed:
fix: FileWidget customRequest support

- Properly support customRequest along with progress updates
- Fix bug when multiple={true} and limit={1} (limit wasn't enforced)
- Fix issue where preview sometimes was not generated when it should have been

davecoates committed:
fix: handle reset with accumulatePages

davecoates committed:
feature: add cleanOnOpen prop

- Default to true so results are more likely fresh
- When dropdown is opened after initial fetch it will clear previous results and
  refetch data

fanglong-alliance commented on js-packages/@prestojs/ui-antd/src/widgets/FileWidget.tsx:
```diff
@@ -161,15 +176,33 @@ export function useFileList(
+                            console.warn(
+                                'No name set for uploaded file. Consider adding a `name` property if passing a `Blob`.'
+                            );
+                            name = `${uid}`;
```
why not valueUid? (as in, this will rely entirely on internal uid instead of using the one from value.uid)

davecoates replied:
good pickup, not intentional

fanglong-alliance reviewed (approved):
looks good otherwise

davecoates committed:
sys: bump antd, fix types/tests

- Require antd 4.4+ now but run tests against 4.9
- Fix various typing and weird test issues with newer antd

davecoates committed:
doc: widget doc updates

davecoates committed:
Merge branch 'fix/async-choices-reset-behaviour'

davecoates committed:
doc: disable search until algolia loaded
